### ISSUE TITLE
lower and upper bounds

### ISSUE THREAD

**Issue by p-lauer** [thouska/spotpy#20]
Hey!

What is the reason, that the upper and lower input parameter bounds are not simply passed from the parameter function to the particular algorithm class but rather are newly calculated by find_min_max? For example see line 55 in sa.py.

As I'm currently commenting and cleaning up my code of an Artificial Bee Colony and a Fitness Scaled Chaotic Artificial Bee Colony implementation, I came to the point where I was looking for a good way to pass the bounds.

From my point of view, you can't reliably reproduce bounds (that were determined in the setup file)  by just generate 1000 random solutions and get the max and min from that, at least in a lot of cases. 

Maybe I am missing a point and there is a quite good reason to use that way.

**Comment by thouska** [comment]
thank you for this valuable comment. It is indeed correct, that the parameter boundaries are determined, by sampling from the prior distribution. Intention was to make the use as simple as possible. Maybe this step is a little oversimplified.
I like your idea to pass the boundaries with the parameter function. I think to assure downwards compatibility, we would need to include the boundaries as an optional argument. If this is not given, than the find_min_max function is used. 
For this we would need to extend the parameter base class, e.g. like this at line 19:
`def __init__(self, name, rndfunc, rndargs, step=None, optguess=None, minbound=None, maxbound=None, *args, **kwargs):`
What do you think?

**Comment by p-lauer** [comment]
Now I see why we need the probabilistic way to find bounds. I'm mostly using an uniform distribution, so I missed that there are other distributions that don't have determined bounds.

I think your solution is good regarding downwards compatibility. Furthermore, we could check if there are bounds passed from the parameter class in find_min_max and depending on that generate probabilistic bounds or use the passed bounds. This would reduce the amount of changes in the algorithms itself.

By the way, I stumbled upon something. It is called Weibull distribution, not Weilbull (line 243 in parameter.py), also it has to be rnd.weibull .

**Comment by thouska** [comment]
I picked up your idea and changed the parameter.py file accordingly. It is now possible to specify the bounds of each parameter by just adding the bounds in the setup file:

`spotpy.parameter.Uniform('x', -10, 10, 1.5, 3.0, -10, 10)`

Alternatively, if the bounds are not given (`spotpy.parameter.Uniform('x', -10, 10, 1.5, 3.0)`), they are determined, by this old find_min_max function (which moved from the algorithm to the parameter class and looks now a little bit nicer). The bounds can be accessed in the algorithm class by using:

`self.parameter()['minbound']` or `self.parameter()['maxbound']`

I changed the implemented algorithms, to use the new style. It is available with version >1.2.26
This fixes also the typo in the Weibull function. Thanks for reporting this!
